main/webapp/modules/core/scripts/index/default-importing-controller/controller.js

```diff
@@ -265,12 +265,14 @@ Refine.DefaultImportingController.prototype.getPreviewData = function(callback,
           "limit" : numRows || 100 // More than we parse for preview anyway
         }),
         null,
-        function(data) {
-          result.rowModel = data;
-          callback(result);
-        },
+		function(data) {
+			  result.rowModel = data;
```

Comment:
I see you removed the error checking here. I think it's a mistake. I think that if the API connection works, we end up here, but still, we might receive no valid data from the back-end, because input data was invalid.

@WaltonG: below (`.fail(function()`), it an error check for an API CONNECTION. Lost of network like. I don't care that we have the same error message, though an error message saying the connection was the problem, could be good. Also, if you check jQuery doc, you'll see that both textStatus and errorThrown can have error information.

Reply:
I moved the error check to xml-parser-ui.js because the error message was specific to xml. We can have the error check back here and generalize the error message to cater for both xml and json

Reply:
> I moved the error check to xml-parser-ui.js because the error message was specific to xml. We can have the error check back here and generalize the error message to cater for both xml and json

That sound logic @WaltonG. Both moving the data error to the function that handles the data, while still checking network & communication errors at the level of the function that initiate the API call. 

So yes, implementing such a check for the JSON (or any code that receives data from this) would be mandatory, since you removed a data check there that was shared.